### ISSUE TITLE
document.querySelector can't find certain elements?

### ISSUE THREAD

**Issue by michael-ziedalski** [puppeteer/puppeteer#4708]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version:
* Platform / OS version: Mac OS 10.11.6
* URLs: https://news.ycombinator.com/
* Node.js version: 12.5.0

**What steps will reproduce the problem?**
I am trying to perform some basic scraping of hacker news (a basic test site), and I cannot figure out how to simply grab some text via `document.querySelector('selector').text()`.  However, I have found out that [`cheerio`](https://github.com/cheeriojs/cheerio), an outside implementation of jquery, unexpectedly produces the expected results :/.

_Please include code that reproduces the issue._

```
const puppeteer = require("puppeteer");
const cheerio = require('cheerio');

async function run(){
  const browser = await puppeteer.launch({
    headless: false,
    args: ["--disable-features=CrossSiteDocumentBlockingAlways,CrossSiteDocumentBlockingIfIsolating"]
  });
  const page = await browser.newPage();
  await page.setViewport({ width: 1600, height: 800 });
  await page.setBypassCSP(true);

  await page.goto('https://news.ycombinator.com/');

  // This works; using cheerio to manually inject jquery into page.
  const html = await page.content();
  const $ = await cheerio.load(html);

  // Successfully printing text of a random article link on page
  await console.log($("*").find("tr#20440105").text());

  // Error here - I have not been able to find a way to 'select' any article link
  // in such a way that this ends up working.
  const text = await page.evaluate(() => document.querySelector("tr#20440105 > td:nth-child(3) > div > a").text();
  )
  await console.log(text);

}

run();
```

**What is the expected result?**
To simply print the text related to an article link. cheerio outputs the following, for a current link: 
`      30.      Seeing France’s Wild Mountains Through a Clouded, Classic Windshield (nytimes.com)`.

**What happens instead?**
I get the following error: 
`UnhandledPromiseRejectionWarning: Error: Evaluation failed: DOMException: Failed to execute 'querySelector' on 'Document': 'tr#20440105 > td:nth-child(3) > div > a' is not a valid selector.`

Any help in getting the default `document.querySelector` to work would be much appreciated, since much of Puppeteer's functionality rests on it. Thanks ahead of time!

**Comment by aslushnikov** [comment]
> and I cannot figure out how to simply grab some text via document.querySelector('selector').text()

@michael-ziedalski two things here:

1. The reason the selector `tr#20440105` does not work for you is  because it is *not* a valid CSS selector - ID's must be CSS [identifiers](https://www.w3.org/TR/CSS21/syndata.html#value-def-identifier), and this is not a CSS identifier.

However, you can match this using the attribute value matching. Instead of `tr#123456`, use `tr[id="123456`

```js
document.querySelector(`tr[id="20440105"] > td:nth-child(3) > div > a` )
```

2. There's no such function as `text()` function on HTMLElement - but you can use `.textContent` better to fetch element's text.

> Any help in getting the default document.querySelector to work would be much appreciated, since much of Puppeteer's functionality rests on it. Thanks ahead of time!

Unlike Cheerio that parses tree itself and implements its own selector engine, Puppeteer simply proxies your calls to the actual browser. 

Hope this helps!

**Comment by Kampmann21** [comment]
Hi @aslushnikov ,
I'm having troubles to convert the CSS selector #startForm\:repeat2\:0\:menuItemLink to a valid selector.
I dropped the hyphen and all backslashes (like mentioned in the specification, which you posted in the comment above), however Puppeteer is still throwing a DOM exception when trying to find that selector.

I tried 

- startForm:repeat2:0:menuItemLink

- startFormrepeat20menuItemLink

Any ideas?

Thanks!
